frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -73,7 +69,7 @@ class DashboardInner extends Component {
 
     if (
       !_.isEqual(prevProps.parameterValues, this.props.parameterValues) ||
-      (!prevProps.dashboard && this.props.dashboard)
```

Comment:
We do initially render Dashboard with `dashboard={null}`. However, this conditional never works as it's superseded by the first one in this method: `prevProps.dashboardId !== this.props.dashboardId`.

Reply:
I was wrong, so reverting this. Won't be committing after working hours anymore.